Paintroid/src/main/java/org/catrobat/paintroid/dialog/SaveInformationDialog.kt

```diff
@@ -0,0 +1,207 @@
+        percentage = jpgView.findViewById(R.id.pocketpaint_percentage_save_info)
+        mySpinner = customLayout.findViewById(R.id.pocketpaint_save_dialog_spinner)
+        informationButton = customLayout.findViewById(R.id.pocketpaint_btn_save_info)
+        imageName = customLayout.findViewById(R.id.pocketpaint_image_name_save_text)
+    }
+
+    private fun initializeFunctioning() {
+        val spinnerArray = arrayListOf("png", "jpg", "ora")
+        val adapter =
+            ArrayAdapter(mySpinner.context, android.R.layout.simple_spinner_item, spinnerArray)
+        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+        mySpinner.adapter = adapter
+        val percentageString = Integer.toString(FileIO.compressQuality)
+        percentage.text = percentageString
+        imageName.text = setName
+        mySeekbar.progress = FileIO.compressQuality
+        mySeekbar.setOnSeekBarChangeListener(this)
+        mySpinner.onItemSelectedListener = this
+        informationButton.setOnClickListener {
+            when {
+                FileIO.isCatrobatImage -> {
+                    presenter.showOraInformationDialog()
+                }
+                FileIO.compressFormat == Bitmap.CompressFormat.JPEG -> {
+                    presenter.showJpgInformationDialog()
+                }
+                else -> {
+                    presenter.showPngInformationDialog()
+                }
+            }
+        }
+    }
+
+    override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
+        when (parent.getItemAtPosition(position).toString()) {
+            "jpg" -> {
+                specificFormatLayout.removeAllViews()
+                specificFormatLayout.addView(jpgView)
+                FileIO.compressFormat = Bitmap.CompressFormat.JPEG
+                FileIO.isCatrobatImage = false
+                FileIO.ending = ".jpg"
+            }
+            "png" -> {
+                specificFormatLayout.removeAllViews()
+                FileIO.compressFormat = Bitmap.CompressFormat.PNG
+                FileIO.isCatrobatImage = false
+                FileIO.ending = ".png"
+            }
+            "ora" -> {
+                specificFormatLayout.removeAllViews()
+                FileIO.compressFormat = Bitmap.CompressFormat.PNG
+                FileIO.isCatrobatImage = true
+                FileIO.ending = ".ora"
+            }
+        }
+    }
+
+    override fun onNothingSelected(parent: AdapterView<*>?) = Unit
+    override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
+        val stringToInsert = "$progress%"
+        percentage.text = stringToInsert
+        FileIO.compressQuality = progress
+    }
+
+    override fun onStartTrackingTouch(seekBar: SeekBar) = Unit
+    override fun onStopTrackingTouch(seekBar: SeekBar) = Unit
+
+    companion object {
+        @JvmStatic
+        fun newInstance(
```

Comment:
The newInstance here is ok. 
You could implement this in a constructor, but I think we would have to provide a standard constructor.
(The bundle is necessary)
https://stackoverflow.com/questions/14011808/why-use-newinstance-for-dialogfragment-instead-of-the-constructor/20953810

Reply:
I've implemented this in a constructor, lemme know how does it look now ;D

Reply:
for reference: The constructor implementation reverted due to possible problems later on.